test/stream.test.js

```diff
@@ -168,3 +168,54 @@ test('Destroying streams prematurely', t => {
     })
   })
 })
+
+test('Destroying streams prematurely with custom error handler set', t => {
+  t.plan(5)
+
+  const fastify = Fastify()
+  const stream = require('stream')
+  const http = require('http')
+
+  fastify.setErrorHandler(function (err, reply) {
+    t.ok(err)
```

Comment:
Can you add a check for errors.code here?

Reply:
`err.code` is `undefined` 🤔 
I changed it to `t.type(err, Error)` though.